trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/VLM.py

```diff
@@ -140,131 +195,213 @@ def VLM(conditions,settings,geometry):
+    TANA   = TNL *(1. - XIA) + TNT *XIA
+    TANB   = TNL *(1. - XIB) + TNT *XIB
+    
+    # cumsum GANT loop if KTOP > 0 (don't actually need KTOP with vectorized arrays and np.roll)
+    GFX    = np.tile((1 /CHORD), (len_mach,1))
+    GANT   = strip_cumsum(GFX*GAMMA, chord_breaks, RNMAX[LE_ind])
+    GANT   = np.roll(GANT,1)
+    GANT[:,LE_ind]   = 0 
+    
+    GLAT   = GANT *(TANA - TANB) - GFX *GAMMA *TANB
+    COS_DL = (YBH-YAH)[LE_ind]/D
+    cos_DL  = np.broadcast_to(np.repeat(COS_DL,RNMAX[LE_ind]),np.shape(B2))
+    DCPSID = FORLAT * cos_DL *GLAT /(XIB - XIA)
+    FACTOR = FORAXL + ONSET
+    
+    
     # COMPUTE LOAD COEFFICIENT
-    GNET = gamma*COSALF*RNMAX/CHORD
-    DCP  = 2*GNET
+    GNET = GAMMA*FACTOR
+    GNET = GNET *RNMAX /CHORD
+    DCP  = 2*GNET + DCPSID
     CP   = DCP
 
     # ---------------------------------------------------------------------------------------
-    # STEP 11: Compute aerodynamic coefficients 
+    # STEP 12: Compute aerodynamic coefficients 
     # ------------------ -------------------------------------------------------------------- 
+    #VORLAX subroutine = AERO
 
     # Work panel by panel
     SURF = np.array(VD.wing_areas)
-    SREF = Sref
-
-    # Unpack coordinates 
-    YAH = VD.YAH*1.
-    ZAH = VD.ZAH
-    ZBH = VD.ZBH    
-    YBH = VD.YBH*1.
-    XA1 = VD.XA1*1.
-    XB1 = VD.XB1*1.
-    YA1 = VD.YA1
-    YB1 = VD.YB1    
-    ZA1 = VD.ZA1
-    ZB1 = VD.ZB1   
-    XA2 = VD.XA2
-    XB2 = VD.XB2      
-    ZA2 = VD.ZA2
-    ZB2 = VD.ZB2    
+    SREF = Sref  
 
     # Flip coordinates on the other side of the wing
     boolean = YBH<0. 
     XA1[boolean], XB1[boolean] = XB1[boolean], XA1[boolean]
     YAH[boolean], YBH[boolean] = YBH[boolean], YAH[boolean]
 
     # Leading edge sweep. VORLAX does it panel by panel. This will be spanwise.
-    TLE = (XB1[LE_ind] - XA1[LE_ind])/ np.sqrt((ZB1[LE_ind]-ZA1[LE_ind])**2 + (YB1[LE_ind]-YA1[LE_ind])**2)
-    
+    TLE   = TAN_LE[:,LE_ind]
     B2_LE = B2[:,LE_ind]
-    T2    = np.broadcast_to(TLE*TLE,np.shape(B2_LE))
+    T2    = TLE*TLE
     STB   = np.zeros_like(B2_LE)
     STB[B2_LE<T2] = np.sqrt(T2[B2_LE<T2]-B2_LE[B2_LE<T2])
-
-    # Panel Dihedral Angle, using AH and BH location
-    D   = np.sqrt((YAH-YBH)**2+(ZAH-ZBH)**2)[LE_ind]
-
-    #SID = ((ZBH-ZAH)[LE_ind]/D) # Just the LE values
-    COD = ((YBH-YAH)[LE_ind]/D) # Just the LE values
+    
+    # DL IS THE DIHEDRAL ANGLE (WITH RESPECT TO THE X-Y PLANE) OF
+    # THE IR STREAMWISE STRIP OF HORSESHOE VORTICES. 
+    COD = np.cos(phi[0,LE_ind])  # Just the LE values
+    SID = np.sin(phi[0,LE_ind])  # Just the LE values
 
     # Now on to each strip
     PION = 2.0 /RNMAX
     ADC  = 0.5*PION
 
     # XLE = LOCATION OF FIRST VORTEX MIDPOINT IN FRACTION OF CHORD.
     XLE = 0.125 *PION
+    
+    GAF = 0.5 + 0.5 *RJTS**2
+
+    # CORMED IS LENGTH OF STRIP CENTERLINE BETWEEN LOAD POINT
+    # AND TRAILING EDGE? THIS PARAMETER IS USED IN THE COMPUTATION
```

Comment:
What's this question from?

Reply:
It's in VORLAX. I'm not sure why...might've been a typo at the time.

Reply:
you don't need to be verbatim, the code is not verbatim